modules/stochastic_tools/src/surrogates/ActiveLearningGaussianProcess.C

```diff
@@ -0,0 +1,116 @@
+  MooseEnum tuning_type("tao adam none", "none");
+  params.addRequiredParam<MooseEnum>(
+      "tuning_algorithm", tuning_type, "Hyper parameter optimizaton algorithm");
+  params.addParam<unsigned int>("iter_adam", 1000, "Tolerance value for Adam optimization");
+  params.addParam<unsigned int>("batch_size", "The batch size for Adam optimization");
+  params.addParam<Real>("learning_rate_adam", 0.001, "The learning rate for Adam optimization");
+  params.addParam<std::string>(
+      "tao_options", "", "Command line options for PETSc/TAO hyperparameter optimization");
+  params.addParam<bool>(
+      "show_optimization_details", false, "Switch to show TAO or Adam solver results");
+  params.addParam<std::vector<std::string>>("tune_parameters",
+                                            "Select hyperparameters to be tuned");
+  params.addParam<std::vector<Real>>("tuning_min", "Minimum allowable tuning value");
+  params.addParam<std::vector<Real>>("tuning_max", "Maximum allowable tuning value");
+  return params;
+}
+
+ActiveLearningGaussianProcess::ActiveLearningGaussianProcess(const InputParameters & parameters)
+  : SurrogateTrainerBase(parameters),
+    CovarianceInterface(parameters),
+    SurrogateModelInterface(this),
+    _gp_handler(declareModelData<StochasticTools::GaussianProcessHandler>("_gp_handler")),
+    _training_params(declareModelData<RealEigenMatrix>("_training_params")),
+    _standardize_params(getParam<bool>("standardize_params")),
+    _standardize_data(getParam<bool>("standardize_data")),
+    _optimization_opts(StochasticTools::GaussianProcessHandler::GPOptimizerOptions(
+        getParam<MooseEnum>("tuning_algorithm"),
+        getParam<std::string>("tao_options"),
+        getParam<bool>("show_optimization_details"),
+        getParam<unsigned int>("iter_adam"),
+        isParamValid("batch_size") ? getParam<unsigned int>("batch_size") : 0,
+        getParam<Real>("learning_rate_adam")))
+{
+  if (isParamValid("batch_size") && _optimization_opts.opt_type == "tao")
+    mooseError("Mini-batch sampling is not compatible with the TAO optimization library. Please "
+               "use Adam optimization.");
+
+  std::vector<std::string> tune_parameters(getParam<std::vector<std::string>>("tune_parameters"));
+  std::vector<Real> bounds;
+  _gp_handler.initialize(
+      getCovarianceFunctionByName(parameters.get<UserObjectName>("covariance_function")),
+      tune_parameters,
+      bounds,
+      bounds);
+}
+
+void
+ActiveLearningGaussianProcess::reTrain(const std::vector<std::vector<Real>> & inputs,
+                                       const std::vector<Real> & outputs) const
+{
+
+  // Addtional error check for each re-train call of the GP surrogate
+  if (isParamValid("batch_size"))
+    if (outputs.size() < _optimization_opts.batch_size)
+      paramError("batch_size", "Batch size cannot be greater than the training data set size.");
```

Comment:
Since `_optimization_opts.batch_size` default to 0, this should work:
```suggestion
  if (outputs.size() < _optimization_opts.batch_size)
    paramError("batch_size", "Batch size cannot be greater than the training data set size.");
```